bricks-maven-plugin/src/main/java/com/github/ingogriebsch/bricks/maven/plugin/analyzer/AnalyzerContext.java

```diff
@@ -0,0 +1,187 @@
+/*-
+ * #%L
+ * Bricks Maven Plugin
+ * %%
+ * Copyright (C) 2018 - 2019 Ingo Griebsch
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.github.ingogriebsch.bricks.maven.plugin.analyzer;
+
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Stopwatch;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.project.MavenProject;
+import org.reflections.Reflections;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Accessors;
+import lombok.experimental.Delegate;
+
+public interface AnalyzerContext extends AutoCloseable {
+
+    void close(); // promises to not throw an Exception!
+
+    Log log();
+
+    MavenProject project();
+
+    Reflections reflections();
+
+    MavenSession session();
+
+    AnalyzerContext childContext(String prefix);
+
+    public static AnalyzerContextImpl of(@NonNull MavenSession session, @NonNull Log log) {
+        return new AnalyzerContextImpl(session, log);
+    }
+}
+
+@Accessors(fluent = true)
+class AnalyzerContextImpl implements AnalyzerContext {
+
+    @Getter
+    private final MavenSession session;
+
+    @Getter
+    private final MavenProject project;
+
+    @Getter
+    private final Log log;
+
+    private final Stopwatch stopWatch;
```

Comment:
Because you are afraid of that it could took years to analyse and you do not realize? :D